### ISSUE TITLE
How to write the `.lp` file of a Ipopt model?

### ISSUE THREAD

**Issue by type-null** [coin-or/CppAD#164]
I am trying to debug the model in CppAD using Ipopt. Is there a way to write the model formulation out as a file, such as .lp file? Thanks

**Comment by bradbell** [comment]
Have looked at ipopt_solve ?
https://cppad.readthedocs.io/en/latest/ipopt_solve.html

**Comment by type-null** [comment]
> Have you looked at ipopt_solve ?
> https://cppad.readthedocs.io/en/latest/ipopt_solve.html

Yes. I'm trying to write a complicated model so I wondered if I can print out the model before solving it to debug. I couldn't find any function in this documentation to write out the model formulation.

**Comment by bradbell** [comment]
The model functions are just a sequence of floating point operations. There is a way to print one out but I do not think it will serve your purpose; see
https://cppad.readthedocs.io/en/latest/to_json.html

Not only that, but you can input a function this way, see
https://cppad.readthedocs.io/en/latest/from_json.html

**Comment by type-null** [comment]
Thanks a lot! I ended up with cout variables of interest to debug...

**Comment by bradbell** [comment]
This issue 164 will be converted to a discussion.
